packages/client/src/modules/play/utils/consumption.spec.ts

```diff
@@ -0,0 +1,631 @@
+import { ActionNames } from "../../../utils/types";
+import { ConsumptionDatum } from "../../persona/consumption";
+import { Persona } from "../../persona/persona";
+import { availableActions } from "../playerActions/constants/actions";
+import { computeNewConsumptionData } from "./consumption";
+
+interface Test {
+  name: string;
+  setup: {
+    consumption: Pick<ConsumptionDatum, "type" | "name" | "value">[];
+    performedActions: ActionNames[];
+  };
+  expected: {
+    consumptionData: Pick<ConsumptionDatum, "type" | "name" | "value">[];
+  };
+}
+
+const runTests = (tests: Test[]) => {
+  tests.forEach((test) => {
+    it(test.name, () => {
+      const consumption: ConsumptionDatum[] = test.setup
+        .consumption as ConsumptionDatum[];
+      const performedActions: ActionNames[] = test.setup.performedActions;
+
+      const newData = computeNewConsumptionData(
+        { consumption } as unknown as Persona,
+        performedActions
+      );
+
+      expect(newData).toEqual(test.expected.consumptionData);
+    });
+  });
+};
+
+describe("consumtion", () => {
+  describe("single action", () => {
+    const TESTS: Test[] = [
+      {
+        name: `should ${availableActions.REDUCE_PLANE_HALF}`,
+        setup: {
+          consumption: [{ type: "fossil", name: "plane", value: 10 }],
+          performedActions: [availableActions.REDUCE_PLANE_HALF],
+        },
+        expected: {
+          consumptionData: [{ type: "fossil", name: "plane", value: 5 }],
+        },
+      },
+      {
+        name: `should ${availableActions.TRANSPORT_STOP_PLANE}`,
+        setup: {
+          consumption: [{ type: "fossil", name: "plane", value: 10 }],
+          performedActions: [availableActions.TRANSPORT_STOP_PLANE],
+        },
+        expected: {
+          consumptionData: [{ type: "fossil", name: "plane", value: 0 }],
+        },
+      },
+      {
+        name: `should ${availableActions.LOCAL_CONSUMPTION}`,
+        setup: {
+          consumption: [
+            { type: "grey", name: "greyOther", value: 10 },
+            { type: "grey", name: "greyTransport", value: 10 },
+          ],
+          performedActions: [availableActions.LOCAL_CONSUMPTION],
+        },
+        expected: {
+          consumptionData: [
+            { type: "grey", name: "greyOther", value: 4.6 },
+            { type: "grey", name: "greyTransport", value: 8.8 },
```

Comment:
@Baboo7 c'est vraiment top que tu écrives des tests, mais ils sont fragiles car on va devoir les changer à chaque changement du modèle. Que penses-tu de tester une inégalité du type `newGreyOther < oldGreyOther` ?

Reply:
@louis-sanna-eki je pense que ça vaut le coup de travailler en valeur exacte pour mieux controler les calculs, notamment l'ordre dans lequel les actions sont traitées. Qu'est ce que vous en pensez @remiriv @Vladou73 ?

Reply:
@Baboo7 ok pour moi.

Reply:
Ça marche 👌

Reply:
@Baboo7 j'avais pas vu mais oui je trouve ça plus pratique d'avoir des vrais tests de valeurs 
après peut être que tout dans un seul fichier c'est un peu dense si on doit modifier les tests de temps en temps, à voir pour les prochains si on fait pas des fichiers plus petits par thème de tests